src/validate-import.ts

```diff
@@ -0,0 +1,30 @@
+import type { Env } from "./env"
+import { normalizeRelativePath } from "./utils"
+import { type EnvMatchers, validateId } from "./validate-id"
+
+export function validateImport({
+  id,
+  denyImports,
+  root,
+  importer,
+  env,
+}: {
+  id: string
+  denyImports: EnvMatchers
+  root: string
+  importer: string
+  env: Env
+}): void {
+  validateId({
+    id,
+    env,
+    invalidIds: denyImports,
+    errorMessage: ({ matcherString }) =>
+      [
+        `Import denied in ${env} environment`,
```

Comment:
If Vite doesn't already prepend the plugin name, could we add it to the error message here and for `validateFile` too? Something like:

```suggestion
        `[vite-env-only] Import denied in ${env} environment`,
```

Reply:
Done.